odoo/addons/base/tests/test_expression.py

```diff
@@ -1504,8 +1786,8 @@ def test_rec_names_search(self):
             SELECT "ir_model"."id", "ir_model"."name"->>%s
             FROM "ir_model"
             WHERE (
-                (("ir_model"."name"->>%s NOT ILIKE %s) OR "ir_model"."name"->>%s IS NULL)
-                AND (("ir_model"."model" NOT ILIKE %s) OR "ir_model"."model" IS NULL)
+                ("ir_model"."model" NOT ILIKE %s OR "ir_model"."model" IS NULL)
+                AND ("ir_model"."name"->>%s NOT ILIKE %s OR "ir_model"."name"->>%s IS NULL)
```

Comment:
Reordering conditions makes debugging harder IMO.  If I am not mistaken, it's a side-effect of the implementation and is not necessary.  What we really want is grouping conditions to enable simplifications and optimizations. I would be less disturbed if the grouping was done by first order of appearance.

Reply:
To be discussed in real life.

IMO sorting by the fields, makes it easier to find the field you are looking for in the SQL.
The ORM generates the SQL and can manipulate the domain at will as long as it is equivalent semantically; it is a liberty for the implementation to choose the order. I have tried grouping by first occurrence and that made the code more complex and slower due to creation of more temporary objects.